.github/workflows/ci-test.yaml

```
@@ -68,13 +70,13 @@ jobs:
     - name: Set up Python
       uses: actions/setup-python@v5
       with:
-        python-version: '3.x'
+        python-version: '3.8'
+
+    - name: Install uv
+      uses: astral-sh/setup-uv@v3
 
     - name: Install dependencies
-      run: |
-        python -m venv venv
-        source venv/bin/activate
-        pip install -r requirements-all.txt
+      run: uv sync --all-extras
```

you do not want to update lock file
```suggestion
      run: uv sync --all-extras --frozen
```